lib/dash/content_protection.js

```diff
@@ -262,6 +262,180 @@ shaka.dash.ContentProtection.parseFromRepresentation = function(
 };
 
 
+/**
+ * Gets a Widevine license URL from a content protection element
+ * containing a custom `ms:laurl` element
+ *
+ * @param {shaka.dash.ContentProtection.Element} element
+ * @return {string}
+ */
+shaka.dash.ContentProtection.getWidevineLicenseUrl = function(element) {
+  const mslaurlNode = shaka.util.XmlUtils.findChild(element.node, 'ms:laurl');
+  if (mslaurlNode) {
+    return mslaurlNode.getAttribute('licenseUrl') || '';
+  }
+  return '';
+};
+
+
+/**
+* @typedef {{
+*   type: number,
+*   value: !Uint16Array
+* }}
+*
+* @description
+* The parsed result of a PlayReady object record.
+*
+* @property {number} type
+*   Type of data stored in the record.
+* @property {!Uint16Array} value
+*   Record content.
+*/
+shaka.dash.ContentProtection.PlayReadyRecord;
+
+/**
+ * Enum for PlayReady record types.
+ * @enum {number}
+ */
+shaka.dash.ContentProtection.PLAYREADY_RECORD_TYPES = {
+  RIGHTS_MANAGEMENT: 0x001,
+  RESERVED: 0x002,
+  EMBEDDED_LICENSE: 0x003,
+};
+
+
+/**
+* Parses a Uint16Array for PlayReady Object Records.
+* Each PRO Record is preceeded by it's PlayReady
+* Record type and length in bytes.
+*
+* PlayReady Object Record format: https://goo.gl/FTcu46
+*
+* @param {!Uint16Array} recordData
+* @return {!Array.<shaka.dash.ContentProtection.PlayReadyRecord>}
+* @private
+*/
+shaka.dash.ContentProtection.parseMsProRecords_ = function(
+  recordData) {
+  let head = 0;
+  let records = [];
+
+  while (head < recordData.length - 1) {
+    const type = recordData[head];
```

Comment:
`Uint16Array` uses the host byte ordering, but I think the PlayReady data always uses the same byte ordering.  You should use `DataView`.

Reply:
Fixed.